com.unity.perception/Editor/Character/CharacterTooling.cs

```diff
@@ -0,0 +1,110 @@
+using System.Collections.Generic;
+using static UnityEngine.Perception.Content.CharacterValidation;
+using System.Linq;
+using UnityEngine.Perception.GroundTruth;
+
+namespace UnityEngine.Perception.Content
+{
+    public class CharacterTooling : MonoBehaviour
+    {
+        /// <summary>
+        /// Bool function used for testing to make sure the target character has the required 15 starting bones
+        /// </summary>
+        /// <param name="selection">target character selected</param>
+        /// <param name="failed">Dictionary return if of Human Bones that tracks they are prsent or missing</param>
+        /// <returns></returns>
+        public bool CharacterRequiredBones(GameObject selection, out Dictionary<HumanBone, bool> failed)
+        {
+            var result = AvatarRequiredBones(selection);
+            failed = new Dictionary<HumanBone, bool>();
+
+            for (int i = 0; i < result.Count; i++)
+            {
+                var bone = result.ElementAt(i);
+                var boneKey = bone.Key;
+                var boneValue = bone.Value;
+
+                if (boneValue != true)
+                    failed.Add(boneKey, boneValue);
+            }
+
+            return failed.Count == 0;
+        }
+
+        /// <summary>
+        /// Ensures there is pose data in the parent and child game objects of a character by checking for pos and rot
```

Comment:
I would go ahead and write out 'position' and 'rotation' since this API comments

Reply:
Fixed in next changeset, will push later today